.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: Node.js CI
+on: [pull_request]
+
+jobs:
+  git_checks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          architecture: "x64"
+      - name: Initialize project
+        run: npm ci
+      - name: Ensure linear-history
+        run: git merge-base --is-ancestor "origin/${GITHUB_BASE_REF}" HEAD
+      - name: Check for fixup or WIP commits
+        run: |
+          ! git log "origin/${GITHUB_BASE_REF}..HEAD" | egrep ' fixup! |wip|Wip|WIP'
+      - name: Run lint
+        run: pnpm lint
```

Comment:
You made use of pnpm but never enabled `pnpm`. `pnpm` requires its own github action:
https://pnpm.io/continuous-integration#github-actions